tests/test_config_user_groups.cpp

```diff
@@ -0,0 +1,369 @@
+    CHECK(og.pubkey().size() == 32);
+    CHECK(og.pubkey_hex() == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    og.priority = 14;
+
+    // The new data doesn't get stored until we call this:
+    groups.set(og);
+
+    auto [to_push, seqno] = groups.push();
+
+    CHECK(seqno == 1);
+
+    // Pretend we uploaded it
+    groups.confirm_pushed(seqno);
+    CHECK(groups.needs_dump());
+    CHECK_FALSE(groups.needs_push());
+
+    session::config::UserGroups g2{seed, groups.dump()};
+    CHECK_FALSE(groups.needs_push());
+    CHECK_FALSE(groups.needs_dump());
+    CHECK(groups.push().second == 1);
+    CHECK_FALSE(groups.needs_dump());  // Because we just called dump() above, to load up g2
+
+    CHECK_FALSE(g2.needs_push());
+    CHECK_FALSE(g2.needs_dump());
+    CHECK(g2.push().second == 1);
+    CHECK_FALSE(g2.needs_dump());
+
+    CHECK(g2.size() == 2);
+    CHECK(g2.size_communities() == 1);
+    CHECK(g2.size_legacy_groups() == 1);
+
+    auto x1 = g2.get_legacy_group(definitely_real_id);
+    REQUIRE(x1);
+    auto& c1 = *x1;
+    CHECK(to_hex(c1.enc_pubkey) == oxenc::to_hex(lg_pk.begin(), lg_pk.end()));
+    CHECK(to_hex(c1.enc_seckey) == oxenc::to_hex(lg_sk.begin(), lg_sk.begin() + 32));
+    CHECK(c1.disappearing_timer == 60min);
+    CHECK(c1.session_id == definitely_real_id);
+    CHECK_FALSE(c1.hidden);
+    CHECK(c1.priority == 3);
+    CHECK(c1.members() == expected_members);
+    CHECK(c1.name == "Englishmen");
+
+    CHECK_FALSE(g2.needs_push());
+    CHECK_FALSE(g2.needs_dump());
+    CHECK(g2.push().second == 1);
+    CHECK_FALSE(g2.needs_dump());
+
+    for (auto* g : {&groups, &g2}) {
+        std::vector<std::string> seen;
+        for (const auto& group : *g) {
+            if (auto* lg = std::get_if<session::config::legacy_group_info>(&group)) {
+                auto [admins, members] = lg->counts();
+                seen.push_back(
+                        "legacy: " + lg->name + ", " + std::to_string(admins) + " admins, " +
+                        std::to_string(members) + " members");
+            } else if (auto* og = std::get_if<session::config::community_info>(&group)) {
+                seen.push_back("community: " + og->base_url() + "/r/" + og->room());
+            } else {
+                seen.push_back("unknown");
+            }
+        }
+
+        CHECK(seen == std::vector<std::string>{
+                              "community: http://example.org:5678/r/SudokuRoom",
+                              "legacy: Englishmen, 1 admins, 2 members",
+                      });
+    }
+
+    auto x2 = g2.get_community("http://EXAMPLE.org:5678", "sudokuRoom");
+    REQUIRE(x2);
+    CHECK(x2->base_url() == "http://example.org:5678");
+    CHECK(x2->room() == "SudokuRoom");  // Case preserved from the stored value, not the input value
+    CHECK(x2->room_norm() == "sudokuroom");
+    CHECK(x2->pubkey_hex() == to_hex(open_group_pubkey));
+    CHECK(x2->priority == 14);
```

Comment:
@jagerman I seem to be getting a `priority` of `0` when retrieving the community via the C library here (I don't think I'm doing anything different...)

Reply:
The legacy group priority check seems to be working fine though (`CHECK(c1.priority == 3)`)